python/semantic_kernel/connectors/memory/astradb/astradb_memory_store.py

```diff
@@ -0,0 +1,300 @@
+# Copyright (c) Microsoft. All rights reserved.
+
+import logging
+from typing import List, Optional, Tuple
+
+from numpy import ndarray
+
+from semantic_kernel.connectors.memory.astradb.astra_client import AstraClient
+from semantic_kernel.connectors.memory.astradb.utils import (
+    build_payload,
+    parse_payload,
+)
+from semantic_kernel.memory.memory_record import MemoryRecord
+from semantic_kernel.memory.memory_store_base import MemoryStoreBase
+
+import aiohttp
+
+MAX_DIMENSIONALITY = 20000
+MAX_UPSERT_BATCH_SIZE = 100
+MAX_QUERY_WITHOUT_METADATA_BATCH_SIZE = 10000
+MAX_QUERY_WITH_METADATA_BATCH_SIZE = 1000
+MAX_FETCH_BATCH_SIZE = 1000
+MAX_DELETE_BATCH_SIZE = 1000
+
+
+class AstraDBMemoryStore(MemoryStoreBase):
+    """A memory store that uses Astra database as the backend."""
+
+    logger: logging.Logger = logging.getLogger(__name__)
+
+    def __init__(
+        self,
+        astra_application_token: str,
+        astra_id: str,
+        astra_region: str,
+        keyspace_name: str,
+        embedding_dim: int,
+        similarity: str,
+        logger: Optional[logging.Logger] = None,
+        session: Optional[aiohttp.ClientSession] = None
+    ) -> None:
+        """Initializes a new instance of the AstraDBMemoryStore class.
+
+        Arguments:
+            astra_application_token {str} -- The Astra application token.
+            astra_id {str} -- The Astra id of database.
+            astra_region {str} -- The Astra region
+            keyspace_name {str} -- The Astra keyspace
+            embedding_dim {int} -- The dimensionality to use for new collections.
+            similarity {str} -- TODO
+            logger {Optional[logging.Logger]} -- The logger to use. (default: {None})
+            session -- Optional session parameter
+        """
+        self._embedding_dim = embedding_dim
+        self._similarity = similarity
+        self._session = session
+
+        if self._embedding_dim > MAX_DIMENSIONALITY:
+            raise ValueError(
+                f"Dimensionality of {self._embedding_dim} exceeds "
+                + f"the maximum allowed value of {MAX_DIMENSIONALITY}."
+            )
+
+        self._client = AstraClient(
+            astra_id=astra_id, astra_region=astra_region, astra_application_token=astra_application_token, keyspace_name=keyspace_name, embedding_dim=embedding_dim, similarity_function=similarity, session= self._session)
+
+    def get_collections(self) -> List[str]:
+        """Gets the list of collections.
+
+        Returns:
+            List[str] -- The list of collections.
+        """
+        return self._client.find_collections(False)
```

Comment:
and this is calling a async method, so this will not work anyway

Reply:
Done